syntax = "proto3";

package unary;

service Unary{
  rpc GetServerResponse(Message) returns (MessageResponse) {}
  rpc GetBooks(Message) returns (stream BooksResponse) {}
  rpc GetBook(Message) returns (BookResponse) {}
  rpc DeleteBook(Message) returns (MessageResponse) {}
  rpc PutBook(MessagePut) returns (MessageResponse) {}
  rpc PostBook(Book) returns (MessageResponse) {}
}

message Book {
  string id = 1;
  string title = 2;
  string author = 3;
  string first_sentence = 4;
  string year_published = 5;
}

message Message{
 string message = 1;
}

message MessagePut{
  string id = 1;
  string key = 2; 
  string value = 3;
}

message MessagePost{
  Book bookPost = 1;
}

message MessageResponse{
 string message = 1;
 bool received = 2;
}

message BookResponse{
 Book book = 1;
}

message BooksResponse{
 Book books = 1;
} 